### PR TITLE
Fix missing output of raw data source. Fixes #2390

### LINKED ISSUE #2390: Sorting no longer draw from raw data
```
{
  "$schema": "https://vega.github.io/schema/vega-lite/v2.json",
  "description": "A scatterplot showing horsepower and miles per gallons for various cars.",
  "data": {"url": "data/cars.json"},
  "mark": "bar",
  "encoding": {
    "x": {"field": "Cylinders","type": "nominal", "sort": {"op": "count", "field": "*", "order": "descending"}},
    "y": {"aggregate": "count","type": "quantitative"}
  }
}
```

vega_editor

### PR DESCRIPTION
**Pull request by domoritz**

### CODE REVIEW

File: src/compile/scale/domain.ts [hunk 1]
```diff
@@ -134,8 +134,7 @@ function parseSingleChannelDomain(scale: Scale, model: UnitModel, channel:Channe
         };
       }
     }
-  } else if (sort) { // have sort -- only for ordinal
-
+  } else if (sort) { // have sort -- only for ordinal and nominal
```
Comment (kanitw) [inline src/compile/scale/domain.ts]: Only for discrete scale (ordinal/point/band)

**Comment by domoritz** [comment]
> You said you are going to add a test to prevent regression.

We have a sample spec. Writing a test for this one case requires new test infra, which is beyond the scope of this pr.

File: src/compile/scale/domain.ts (hunk 1 above)
Comment (domoritz) [inline src/compile/scale/domain.ts]: I deleted the comment to avoid confusion.

**Comment by kanitw** [comment]
Please file an issue and merge this

**Comment by domoritz** [comment]
Follow up issue about testing: https://github.com/vega/vega-lite/issues/2399

**Pull request merged**